Making current folder persist across page changes.

webinista opened the pull request:
- Updates state with an `inFolder` property when the user navigates to a new folder.
- Sets the `usa-active` class when the value of `inFolder` matches the current folder id.
- Makes /messaging/folders/ redirect to the Inbox.

WIP on department-of-veterans-affairs/messaging-fe#69
Also affects department-of-veterans-affairs/messaging-fe#44

webinista committed:
Making current folder persist across page changes.

- Updates state with an `inFolder` property when the user navigates to a new folder.
- Sets the `usa-active` class when the value of `inFolder` matches the current folder id.
- Makes /messaging/folders/ redirect to the Inbox.

WIP on department-of-veterans-affairs/messaging-fe#69
Also affects department-of-veterans-affairs/messaging-fe#44

webinista committed:
Making current folder persist across page changes.

- Updates state with an `inFolder` property when the user navigates to a new folder.
- Sets the `usa-active` class when the value of `inFolder` matches the current folder id.
- Makes /messaging/folders/ redirect to the Inbox.

WIP on department-of-veterans-affairs/messaging-fe#69
Also affects department-of-veterans-affairs/messaging-fe#44

webinista committed:
Making current folder persist across page changes.

- Updates state with an `inFolder` property when the user navigates to a new folder.
- Sets the `usa-active` class when the value of `inFolder` matches the current folder id.
- Makes /messaging/folders/ redirect to the Inbox.

WIP on department-of-veterans-affairs/messaging-fe#69
Also affects department-of-veterans-affairs/messaging-fe#44

webinista committed:
Making current folder persist across page changes.

- Updates state with an `inFolder` property when the user navigates to a new folder.
- Sets the `usa-active` class when the value of `inFolder` matches the current folder id.
- Makes /messaging/folders/ redirect to the Inbox.

WIP on department-of-veterans-affairs/messaging-fe#69
Also affects department-of-veterans-affairs/messaging-fe#44

U-DON commented on src/js/messaging/reducers/folders.js:
```diff
@@ -55,6 +57,9 @@ export default function folders(state = initialState, action) {
       return set('ui.nav.foldersExpanded', !state.ui.nav.foldersExpanded, state);
+    case SET_CURRENT_FOLDER:
+      // The + forces +action.folderId to be a number
+      return set('inFolder', +action.folderId, state);
```
I'd like to see it set as `data.currentItem.id` or something like that, as long as its associated with `currentItem`. And then when the fetch folder succeeds, the result can be `assign`ed instead of using `set`.

U-DON commented on src/js/messaging/components/FolderNav.jsx:
```diff
@@ -4,14 +4,12 @@ import classNames from 'classnames';
     this.makeFolderLink = this.makeFolderLink.bind(this);
     this.makeMyFolders = this.makeMyFolders.bind(this);
   }
-
```
Bring back this newline?

U-DON reviewed (changes requested):
The way you did it LGTM, but I think the current folder id should be namespaced under the `currentItem` in the store.

U-DON committed:
[Messaging] Updated MessageDetails component to work on mobile. (#3149)

* Refactored how message details appear. Uses CSS and events instead of React state.

* Mobile version of message details.

* Hide 'From' and 'Subject Line' from message details in mobile.

* Time format in message details.

* Use const for variable.

* Removed obsolete import.

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#69-persist-folders' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#69-persist-folders

webinista committed:
Renaming rx/_partials to rx/partials. Closes #3087 (#3151)

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#69-persist-folders' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#69-persist-folders

jbalboni committed:
Add scrolling back to edu benefits (#3145)

webinista committed:
Moving persistFolder property into folders object.

- Renamed from inFolder to persistFolder
- Maked persistFolder a child property of currentItem.

webinista committed:
Fixing merge conflicts.
